### ISSUE TITLE
error: cjs / mjs files are compiled as TypeScript

### ISSUE THREAD

**Issue by taisukef** [denoland/deno#6176]
Receiving TS2531 [ERROR] when running the mjs/cjs after Deno 1.0.4.

save as "test.mjs"
```
import { exp } from "./test_mod.ts";
if (false) null.func();
```
save as "test_mod.ts"
```
export const exp = {};
```
and run
```
% deno run test.mjs
Compile file:/// ... /test.mjs
error: TS2531 [ERROR]: Object is possibly 'null'.
if (false) null.func();
           ~~~~
    at file:/// ... /test.mjs:2:12
```

1.0.5 -- NG
1.0.4 -- NG
1.0.3 -- OK

without ts import -- OK
import mjs -- OK

**Comment by bartlomieju** [comment]
@taisukef since you got `.ts` import in `test.mjs` it will be run through TS compiler. That said - it's a bit surprising that error is reported even though `checkJs` setting is off. CC @kitsonk do you know what might be the problem?

**Comment by kitsonk** [comment]
@bartlomieju only thing I can think of is that the media type is off: https://github.com/denoland/deno/blob/master/cli/js/compiler.ts#L171-L190

I didn't think that the compiler looked at anything other than that.

**Comment by nsadeh** [comment]
Hello I got re-routed to this issue through #7282. Do we know the status for this issue? Thanks :)

**Comment by kitsonk** [comment]
It is open and unresolved.

**Comment by taisukef** [comment]
I use "js" instead of "mjs" to avoid this err.

**Comment by kitsonk** [comment]
The original error is resolved in 1.5, but the problem is now that `tsc` doesn't support a root module with an `.mjs` extension and so we need to update our logic to "hide" this from `tsc` like we do for other ones.  Should be a trivial fix.
